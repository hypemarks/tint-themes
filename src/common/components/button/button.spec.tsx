import { render, screen } from '@testing-library/react';
import React from 'react';
import Button from './button';
import userEvent from '@testing-library/user-event';

describe('Test <Button /> Component', () => {
  it('should match snapshot', () => {
    const wrapper = render(<Button href={'tintup.com'} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should create button with passed params', () => {
    const className = 'className';
    const style = { color: 'red' };
    const wrapper = render(<Button className={className} style={style} disabled={false} type={'reset'} />);
    const button = wrapper.queryByTestId('button');
    expect(button).toBeInTheDocument();
    expect(button).toHaveStyle(style);
    expect(button).toHaveClass(className);
  });

  it('should called onClick callback', () => {
    const onclick = jest.fn();
    render(
      <Button onClick={onclick} type={'button'}>
        Fire me
      </Button>
    );
    userEvent.click(screen.getByText('Fire me'));
    expect(onclick).toHaveBeenCalledTimes(1);
  });
});
