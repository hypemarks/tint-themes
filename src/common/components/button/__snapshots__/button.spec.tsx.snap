// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Button /> Component should match snapshot 1`] = `
<DocumentFragment>
  <a
    class=" button"
    href="tintup.com"
    tabindex="0"
  />
</DocumentFragment>
`;
